UniversalAlgebra/Algebras/Setoid.lagda

```diff
@@ -78,5 +84,49 @@ record SetoidAlgebra α ρ : Type (𝓞 ⊔ 𝓥 ⊔ lsuc (α ⊔ ρ)) where
      --       1. a function  f : (⟦ 𝑆 ⟧s Den) .Carrier  → Den . Carrier
      --       2. a proof cong : f Preserves _≈₁_ ⟶ _≈₂_ (that f preserves the setoid equalities)
 
+
+\end{code}
+
+#### Products of SetoidAlgebras
+
+\begin{code}
+
+module _ {α ρ ι : Level} where
+
+ open SetoidAlgebra
+
+ ⨅ : {I : Type ι }(𝒜 : I → SetoidAlgebra α ρ) → SetoidAlgebra _ _ -- (𝓘 ⊔ α) 𝑆
+ open IsEquivalence renaming ( refl  to  reflE
+                             ; sym   to  symE
+                             ; trans to  transE )
+
+ Den (⨅ {I} 𝒜) .∣_∣ = ∀ i → ∣ Den (𝒜 i) ∣
+ Den (⨅ {I} 𝒜) ._≈_ = λ as bs → ∀ i → Den (𝒜 i) ._≈_ (as i) (bs i)
+ Den (⨅ {I} 𝒜) .isEquivalence .reflE = λ i → Den (𝒜 i) .isEquivalence .reflE
+ Den (⨅ {I} 𝒜) .isEquivalence .symE = λ x i → Den (𝒜 i) .isEquivalence .symE (x i)
+ Den (⨅ {I} 𝒜) .isEquivalence .transE = λ x y i → Den (𝒜 i) .isEquivalence .transE (x i) (y i)
+
+ apply (den (⨅ {I} 𝒜)) (f , a) i = apply (den (𝒜 i)) (f , (λ x → a x i))
```

Comment:
```agda
apply (den (𝒜 i)) (f , flip a i)
```
might also appeal to you?

A few combinators to do pointwise lifting are useful. Guillaume Allais has programmed a bunch in stdlib, if only I could find then again...

Reply:
Yes, that's very nice. I had been planning to roll my own such combinator, but hadn't gotten around to it.  I'm glad to know now that it exists and what it's called (flip).  Thanks!